src/movies.js

```diff
@@ -1,25 +1,111 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    let directorsName = [];
+    directorsName = moviesArray.map((element) => {
+        return element.director;
+    });
+    return directorsName;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray){
+    let dramaMoviesSpielberg = [];
+    dramaMoviesSpielberg = moviesArray.filter(element => {
+        if (element.director === 'Steven Spielberg' && element.genre.includes ('Drama')){
+            return element;
+        }
+
+    });
+    return dramaMoviesSpielberg.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (moviesArray.length === 0) return 0;
 
-// Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+    let totalValue = moviesArray.reduce((acc, films) => {
+        if(films.score){
+            return films.score + acc;
+        } else {
+            return acc; 
+        } 
+    },0)
 
-// Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+    return Math.round((totalValue * 100) /moviesArray.length) / 100;
+}
 
+// Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesScore(moviesArray) {
+    let scoresDramaArray = [];
+    let avg = 0;
+    let dramaCounter = 0;
+  
+    scoresDramaArray = moviesArray.filter((movie) => {
+      return movie.genre.includes("Drama");
+    });
+  
+    scoresDramaArray = scoresDramaArray.map((movie) => {
+      return movie.score;
+    });
+  
+    let total = scoresDramaArray.reduce((acc, film) => {
```

Comment:
Previously, you wrote a function that calculates the average scores, right? So, here, you should reuse it. Instead of writing again the logic to calculate the average, you can call the `scoresAverage` function.
`return scoresAverage(scoresDramaArray)`